x-pack/plugins/fleet/server/services/agents/agentless_agent.ts

```diff
@@ -155,7 +163,112 @@ class AgentlessAgentService {
+        cert: tlsConfig.certificate,
+        key: tlsConfig.key,
+        ca: tlsConfig.certificateAuthorities,
+      }),
+    };
+
+    const requestConfigDebug = JSON.stringify({
+      url: getDeletionEndpointPath(agentlessConfig, `/deployments/${agentlessPolicyId}`),
+      method: 'DELETE',
+      headers: {
+        'Content-type': 'application/json',
+      },
+      httpsAgent: new https.Agent({
+        rejectUnauthorized: tlsConfig.rejectUnauthorized,
+        cert: tlsConfig.certificate,
+        key: tlsConfig.key,
+        ca: tlsConfig.certificateAuthorities,
+      }),
+    });
+
+    logger.debug(
+      `[Agentless API] Start deleting agentless agent for agent policy ${requestConfigDebug}`
+    );
+
+    if (!isAgentlessApiEnabled) {
+      logger.error(
+        '[Agentless API] Agentless API is not supported. Deleting agentless agent is not supported in non-cloud or non-serverless environments'
+      );
+    }
+
+    if (!agentlessConfig) {
+      logger.error('[Agentless API] kibana.yml is currently missing Agentless API configuration');
+    }
+
+    logger.debug(`[Agentless API] Deleting agentless agent with TLS config with certificate`);
+
+    logger.debug(
+      `[Agentless API] Deleting agentless deployment with request config ${JSON.stringify({
+        ...requestConfig,
+        httpsAgent: {
+          ...requestConfig.httpsAgent,
+          options: {
+            ...requestConfig.httpsAgent.options,
+            cert: requestConfig.httpsAgent.options.cert ? 'REDACTED' : undefined,
+            key: requestConfig.httpsAgent.options.key ? 'REDACTED' : undefined,
+            ca: requestConfig.httpsAgent.options.ca ? 'REDACTED' : undefined,
+          },
+        },
+      })}`
+    );
+
+    const response = await axios(requestConfig).catch((error: AxiosError) => {
+      const errorLogCodeCause = `${error.code}  ${this.convertCauseErrorsToString(error)}`;
+
+      if (!axios.isAxiosError(error)) {
+        logger.error(
+          `[Agentless API] Deleting agentless deployment failed with an error ${JSON.stringify(
+            error
+          )} ${JSON.stringify(requestConfigDebug)}`
+        );
+      }
+      if (error.response) {
+        logger.error(
+          `[Agentless API] Deleting Agentless deployment Failed Response Error: ${JSON.stringify(
+            error.response.status
+          )}} ${JSON.stringify(error.response.data)}} ${requestConfigDebug} `
+        );
+      } else if (error.request) {
+        logger.error(
+          `[Agentless API] Deleting agentless deployment failed to receive a response from the Agentless API ${errorLogCodeCause} ${JSON.stringify(
+            requestConfigDebug
+          )}`
+        );
+      } else {
+        logger.error(
+          `[Agentless API] Deleting agentless deployment failed to delete the request ${errorLogCodeCause} ${JSON.stringify(
+            requestConfigDebug
+          )}`
+        );
+      }
+    });
```

Comment:
Not for now but when we make changes for the error handling from the Agentless-API we should think about creating a common function